legion/engine/physics/physx/systems/physx_physics_system.hpp

```diff
@@ -12,18 +17,39 @@ namespace legion::physics
     {
     public:
 
+        PhysXPhysicsSystem();
+        ~PhysXPhysicsSystem();
+
         virtual void setup();
 
         void fixedUpdate(time::time_span<fast_time> deltaTime);
 
     private:
 
-        void setupPhysXGlobalVariables();
+        struct physXStatics
+        {
+            bool m_isInit = false;
+
+            physx::PxFoundation* foundation = nullptr;
+            physx::PxPvd* pvd = nullptr;
+            physx::PxDefaultCpuDispatcher* dispatcher = nullptr;
+        };
```

Comment:
if you forward declare `physx_statics` here then you can define the definition in the .cpp file. since i don't think anything in this header needs to know about these implementation details.

```
// hpp
class PhysxSystem final : public System<PhysxSystem>
{
    struct physx_statics;
};
```
```
// cpp
struct PhysxSystem::physx_statics
{
    static size_type systemInstances;
    static physx::PxFoundation* foundation = nullptr;
    static physx::PxPvd* pvd = nullptr;
    static physx::PxDefaultCpuDispatcher* dispatcher = nullptr;
}
```